identity-hub-client/src/main/java/org/eclipse/dataspaceconnector/identityhub/client/ApiException.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.client;
+
+import okhttp3.Headers;
+import okhttp3.ResponseBody;
+
+/**
+ * ApiException. Exception thrown by the IdentityHub client when IdentityHub does not return a 200.
```

Comment:
nit: It is also used when we have empty response e.g. getVerifiableCredentials. So it is not only for not 200 OK responses.

```suggestion
 * Base exception for IdentityHub rest client.
 * The client should use unchecked exceptions when appropriate (e.g., non-recoverable errors) and may extend this exception.
```